SwiftLeeds/Data/Model/Schedule.swift

```diff
@@ -0,0 +1,121 @@
+        }
+
+        static var timeFormat: DateFormatter = {
+            let dateFormatter = DateFormatter()
+            dateFormatter.dateFormat = "HH:mm"
+            return dateFormatter
+        }()
+
+        static let sample = Schedule.Slot(id: UUID(), startTime: "11:00", duration: 30, activity: nil, presentation: .init(id: UUID(), title: "My Title", synopsis: "My synopsis", speaker: .init(id: UUID(), name: "Matthew Gallagher", biography: "Creates apps", profileImage: "", organisation: "PDA Monkey", twitter: "pdamonkey"), image: nil))
+    }
+}
+
+// MARK: - Slot Decodable
+extension Schedule.Slot: Decodable {
+    init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: CodingKeys.self)
+
+        id = try values.decode(UUID.self, forKey: .id)
+        duration = try values.decode(Int.self, forKey: .duration)
+        startTime = try values.decode(String.self, forKey: .startTime)
+
+        if let activity = try values.decodeIfPresent(Activity.self, forKey: .activity) {
+            self.activity = activity
+            self.presentation = nil
+        } else if let presentation = try values.decodeIfPresent(Presentation.self, forKey: .presentation) {
+            self.activity = nil
+            self.presentation = presentation
+        } else {
+            throw(SlotError.invalidSlot)
+        }
+    }
+
+    enum SlotError: Error {
+        case invalidSlot
+    }
+}
+
+// MARK: - Slot Equatable
+extension Schedule.Slot: Equatable {
+    static func == (lhs: Schedule.Slot, rhs: Schedule.Slot) -> Bool {
+        lhs.id == rhs.id
+    }
+}
+
+// MARK: - Activity
+struct Activity: Decodable, Identifiable {
+    let id: UUID
+    let title: String
+    let subtitle: String?
+    let description: String?
+    let image: String?
+    let metadataURL: String?
+
+    init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: CodingKeys.self)
+
+        id = try values.decode(UUID.self, forKey: .id)
+        title = try values.decode(String.self, forKey: .title)
+        subtitle = try values.decodeIfPresent(String.self, forKey: .subtitle)
+        description = try values.decodeIfPresent(String.self, forKey: .description)
+        image = try values.decodeIfPresent(String.self, forKey: .image)
+        metadataURL = try values.decodeIfPresent(String.self, forKey: .metadataURL)
+    }
+
+    private enum CodingKeys: CodingKey {
+        case id, title, subtitle, description, image, metadataURL
+    }
+}
+
+// MARK: - Presentation
+struct Presentation: Decodable, Identifiable {
+    let id: UUID
+    let title: String
+    let synopsis: String
+    let speaker: Speaker?
+    let image: String?
+}
+
+// MARK: - Speaker
+struct Speaker: Decodable, Identifiable {
+    let id: UUID
+    let name: String
+    let biography: String
+    let profileImage: String
+    let organisation: String
+    let twitter: String?
+    //let presentations: [String]?
```

Comment:
Can we remove this instead of commenting it out?

Reply:
You mentioned it might be required for other areas, but as the API doesn't currently provide it I left this in as a placeholder, but happy to remove it